### PR TITLE
Equil constructor

### LINKED ISSUE #36: Specify initial guess / restarting
We should have an easy way to either specify an initial guess, or to restart from a previous run at a certain point.

### LINKED ISSUE #80: New constuctors for equilibria + family
Rather than requiring a dict of inputs, allow Equilibrium to be initialized via Surface + Profile objects

### PR DESCRIPTION
**Pull request by f0uriest**
Updates Equilibrium to make creating them more straightforward.

- Instead of a dictionary of arrays and values, init method now takes individual arguments. These can either be objects of the correct type (ie `Surface` objects for boundary condiitons, `Profile` for pressure and iota etc,) or ndarrays which will get parsed into objects of the correct type (for backwards compatibility)
- Also introduces more options for generating initial guesses, and a new dedicated method `equilibrium.set_initial_guess()`. The default is to scale the boundary surface that is assigned to the equilibrium, but another surface (and axis) can be supplied as an argument to the function to use that surface instead for the initial guess. It also accepts another Equilibrium instance, or a path to a saved DESC or VMEC equilibrium which will be loaded and its flux surfaces will be used as the initial guess.
- Command line interface updated to allow for initial guesses from DESC or VMEC solutions using `--guess=path` (this also replaces the old `--vmec` flag)

Resolves #36
Resolves #80

### CODE REVIEW

**Review by dpanici — APPROVED** [review]
Looks good to me, added some minor comments but it is really useful

File: desc/configuration.py [hunk 1]
```diff
@@ -51,6 +54,39 @@ class _Configuration(IOAble, ABC):
...
+        Default is a FourierRZToroidalSurface with major radius 10 and
+        minor radius 1
+    axis : Curve or ndarray shape(k,3) (optional)
+        Initial guess for the magnetic axis as a Curve object or ndarray
+        of mode numbers and spectral coefficints of the form [n, R, Z].
+        Default is the centroid of the surface.
+    sym : bool (optional)
+        Whether to enforce stellarator symmetry. Default surface.sym or False.
+    spectral_indexing : str (optional)
+        Type of Zernike indexing scheme to use. Default ``'fringe'``
```
[the remark this answers is not in the document]
Comment (f0uriest) [reply desc/configuration.py:88]: good point, will do

File: desc/configuration.py [hunk 2]
```diff
@@ -233,6 +233,331 @@ def _set_basis(self):
...
+    # TODO: make this kwargs instead?
+    def set_initial_guess(self, *args):
+        """Set the initial guess for the flux surfaces, eg R_lmn, Z_lmn, L_lmn
+
+        Parameters
+        ----------
+        args :
+            either:
+              - No arguments, in which case eq.surface will be used
+              - Another Surface object which will be scaled to generate the initial guess
```
[the remark this answers is not in the document]
Comment (f0uriest) [reply desc/configuration.py:388]: this is a really good idea, thanks. I've been meaning to include more example snippets in docstrings and this is a great place to start

File: desc/equilibrium.py [hunk 3]
```diff
@@ -403,12 +476,9 @@ def optimizer(self, optimizer):
     @property
     def initial(self):
         """Return initial equilibrium state from which it was solved (Equilibrium)."""
-        p_modes = np.array(
-            [self.pressure.basis.modes[:, 0], self.p_l, np.zeros_like(self.p_l)]
-        ).T
-        i_modes = np.array(
-            [self.iota.basis.modes[:, 0], np.zeros_like(self.i_l), self.i_l]
-        ).T
+        # TODO: fix this with new x etc
```
[the remark this answers is not in the document]
Comment (f0uriest) [reply desc/equilibrium.py]: not sure actually, i think it might have been an old comment

**Pull request merged**